azuread/resource_application.go

```diff
@@ -307,6 +307,10 @@ func resourceApplicationCreate(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	if err := graph.ApplicationValidateRolesScopes(d.Get("app_role"), d.Get("oauth2_permissions")); err != nil {
```

Comment:
Given this is kinda particular to the schema of this resource i think it might make sense to have it as a local funciton?

Reply:
Thanks, makes sense 👍